.github/workflows/continuous-integration-workflow.yml

```diff
@@ -124,3 +126,20 @@ jobs:
           cmake -B builddir -DCMAKE_CXX_COMPILER=${{ matrix.cxx }}
           cmake --build builddir
           cmake --build builddir --target test
+      - name: Gather benchmark results
+        run: |
+          mkdir ${{ env.BUILD_ID }}
+          find builddir/core/perf_test/ -name "*.json" -exec mv {} ${{ env.BUILD_ID }}/  \;
+      - name: Push benchmark results
+        if: |
+          github.ref == 'refs/heads/performance-results-visualization' &&
+          matrix.distro == 'ubuntu:latest'
```

Comment:
TODO: decide what set of conditions should be used here (which builds should report their results)

Reply:
Use a separate `.yml` file dedicated to benchmarks and decide which build / configuration to use there.